### ISSUE TITLE
[BST] Pet claiming and ability bug

### ISSUE THREAD

**Issue by greyenvy** [EdenServer/community#1596]
### Checklist

- [x] I have searched for existing issues for issues like this one. The issue has not been posted. (Duplicate reports slow down development.)
- [X] I have provided reproducable steps. (No "as the title says" posts please. Provide reproducable steps even if it seems like a waste of time.)
- [X] I have provided my client version in the details. (type /ver into your game window)

### Details
There are two issues I've noticed about pet claiming/aggro:

1) If you begin a fight using pet but use "Sic" or a "Ready" ability before a melee attack happens, the ability does not claim the prey.  It is not until your pet actually attempts a melee attack that the mob's name turns red.  Example, use "Fight" on a mob, while CrabFamiliar is on his way to mob, using "Big Scissors", CrabFamiliar opens up with Big Scissors but prey's name stays White (unclaimed).  Melee attack happens > prey name turns Red.

2) Similarly, if you use a non-damaging ability (ie. Sonic Boom from triple bats) and it hits any targets other than your current prey, they do not aggro.  This makes it too safe for BST to simply use "Sic" or "Ready" abilities.  If Sonic Boom or some other non-damaging AoE ability hits outside mobs, they should all aggro your pet and currently they do not.

**Comment by Hatberg** [comment]
This bug also occurs for SMN. Carbuncle opening with Poison Nails will not claim the mob.

**Comment by gweivyth** [comment]
This is the same when Automaton use ranged attacks as the first action, it doesn't claim the monster until a melee auto attack is dealt.

**Comment by Volkaru** [comment]
This is retail behavior. A pet would not claim a monster until a regular melee attack was performed on it.

**Comment by greyenvy** [comment]
I knew at least that second one was incorrect.  Thank you for your input!

**Comment by gweivyth** [comment]
> As far as the claiming, this is retail behavior. A pet would not claim a monster until a regular melee attack was performed on it.
> Seems AoE abilities not aggroing other mobs is unintended, though.

What about spells?  I haven't messed with the mage frame very much but does Dia from the automaton claim properly?

**Comment by Volkaru** [comment]
Spells nor ranged attacks from Automatons should claim. Siege worked partly on the code to implement ADD properly so you could skill up without getting aggro on master. I can't remember if he did ranged attacks or spells before he took his break, though.

**Comment by KnowOne134** [comment]
all new claiming code installed now, open a new issue with details if reoccurs
